docs: putting test into each chapter of the tutorial (#1647)
canonical/operator: docs/tutorial/from-zero-to-hero-write-your-first-kubernetes-charm/make-your-charm-configurable.md

Congratulations, you now know how to make your charm configurable!

> ## Add unit tests for your charm

Reviewer: Since we've already done the general setup in the previous chapter, I think this title can be more direct:

```suggestion
## Write unit tests
```

> Since we added a new feature to configure `server-port` and use it in the `_pebble_layer` dynamically, we want to add some test cases to cover it. We can add a test case where we set the port in the input state, and assert the port is indeed used in the service's command in the container layer:

Reviewer: ```suggestion
Since we added a new feature to configure `server-port` and use it in the `_pebble_layer` dynamically, we should write tests for the feature.

First, we'll add a test that sets the port in the input state and asserts that the port is used in the service's command in the container layer:
```

```python
def test_config_changed():
    ctx = testing.Context(FastAPIDemoCharm)
    container = testing.Container(name="demo-server", can_connect=True)
    state_in = testing.State(
        containers={container},
        config={"server-port": 8080},
        leader=True,
    )
    ctx.run(ctx.on.config_changed(), state_in)
    assert "--port=8080" in container.layers["fastapi_demo"].services["fastapi-service"].command
```

> In `_on_config_changed`, we specifically don't allow port 22 to be used. When port is configured to 22, we will set the status as blocked. So, we can also add a test case to cover this behaviour by setting the port explicitly to 22 in the input state and assert the unit status is blocked:

Reviewer: ```suggestion
In `_on_config_changed`, we specifically don't allow port 22 to be used. If port 22 is configured, we set the unit status to `blocked`. So, we can add a test to cover this behaviour by setting the port to 22 in the input state and asserting that the unit status is blocked:
```

```python
def test_config_changed_invalid_port():
    ctx = testing.Context(FastAPIDemoCharm)
    container = testing.Container(name="demo-server", can_connect=True)
    state_in = testing.State(
        containers={container},
        config={"server-port": 22},
        leader=True,
    )
    state_out = ctx.run(ctx.on.config_changed(), state_in)
    assert state_out.unit_status == testing.BlockedStatus(
        "Invalid port number, 22 is reserved for SSH"
    )
```

> Run `tox -e unit` to make sure all test cases pass.

Reviewer: ```suggestion
Run `tox -e unit` to check that all tests pass.
```
